src/wasm-lib/kcl/src/parsing/token/mod.rs

```diff
@@ -190,6 +191,21 @@ impl<'a> TokenSlice<'a> {
             stream: self.stream,
         }
     }
+
+    pub fn as_source_range(&self) -> SourceRange {
+        let stream_len = self.stream.tokens.len();
```

Comment:
Originally this was just 
```
let first_token = self.token(0);
let last_token = self.token(self.end-self.start);
```
but I kept getting index out of bounds errors. so it seems like there might be a counting error somewhere, but I couldn't figure out where that might be.